lang/renaming/src/ctx.rs

```diff
@@ -1,54 +1,52 @@
 use syntax::ast::*;
-use syntax::ctx::{Context, ContextElem};
+use syntax::ctx::{Context, ContextElem, GenericCtx};
 
 use super::util::increment_name;
 
-#[derive(Debug, Clone)]
 pub struct Ctx {
-    bound: Vec<Vec<Ident>>,
+    ctx: GenericCtx<Ident>,
+}
+
+impl From<GenericCtx<Ident>> for Ctx {
+    fn from(value: GenericCtx<Ident>) -> Self {
+        Ctx { ctx: value }
+    }
 }
 
 impl Context for Ctx {
     type Elem = Ident;
 
     fn push_telescope(&mut self) {
-        self.bound.push(vec![]);
+        self.ctx.bound.push(vec![]);
     }
 
     fn pop_telescope(&mut self) {
-        self.bound.pop().unwrap();
+        self.ctx.bound.pop().unwrap();
     }
 
     fn push_binder(&mut self, elem: Self::Elem) {
         assert!(elem == "_" || elem.is_empty() || !self.contains_name(&elem));
-        self.bound.last_mut().expect("Cannot push without calling level_inc_fst first").push(elem);
+        self.ctx
+            .bound
+            .last_mut()
+            .expect("Cannot push without calling level_inc_fst first")
```

Comment:
I think `level_inc_fst` here is outdated.

Reply:
Yes, it probably should be `push_telescope`. Changed in [eaebbdb](https://github.com/polarity-lang/polarity/pull/264/commits/eaebbdb50d46591306ed66e1c95c1e26f1db0970)